### DISCUSSION TITLE
setup fail solved, now the nextcloud container is missing permissions

### DISCUSSION THREAD

**Discussion by Kuchenmampfer** [nextcloud/all-in-one#645] [Questions] [answered]
Hi,
I tried to setup nextcloud AIO behind a Caddy reverse proxy. It runs fine and I can access the AIO login on `x.x.x.x:8080`, but when I type my domain (`nextcloud.domain.de`) in that field, I get "Domain does not point to this server or reverse proxy not configured correctly."

I have these lines in my Caddyfile:
```
https://nextcloud.domain.de:443 {
    header Strict-Transport-Security max-age=31536000;
    reverse_proxy localhost:11000
}
```
and I ran nextcloud with:
```
sudo docker run -it --name nextcloud-aio-mastercontainer --restart always
-p 8080:8080 -e APACHE_PORT=11000
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config
--volume /var/run/docker.sock:/var/run/docker.sock:ro nextcloud/all-in-one:latest
```
I have an A record for the nextcloud subdomain pointing to my vps, but when I try to reach it in my browser, it tells me, that the certificate is only valid for `autoconfig.domain.de`, `autodiscover.domain.de` and `mail.domain.de`. Trying to reach `nextcloud.domain.de:8443` just throws an error and my browser wants me to check my internet connection.
I have added iptables rules to open the ports 8443, 3478, https, http and http-alt, but the first two of them do not show up when I run `sudo lsof -i`.

What can I do to get this to work?
Thank you very much for your help <3

**Comment by szaimen** [comment, marked as answer]
Hello, did you already follow https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#4-how-to-debug-things?

**Comment by Kuchenmampfer** [comment]
Kind of.
1. I first used port 11443 and only changed to 11000 when the former did not work.
2. The proxy does run on the host OS (debian 11)
3. I did not look into the logs, but I assume that it has already spawned `nextcloud-aio-domaincheck` because `sudo docker ps` returns
```
e4aee8706881   nextcloud/all-in-one:latest        "start.sh /usr/bin/s…"   5 days ago    Up 5 days   80/tcp, 8443/tcp, 0.0.0.0:8080->8080/tcp, :::8080->8080/tcp                                                                                                                                                                         nextcloud-aio-mastercontainer
3c24489c9873   nextcloud/aio-domaincheck:latest   "/start.sh"              5 days ago    Up 5 days   0.0.0.0:11443->11443/tcp, :::11443->11443/tcp                                                                                                                                                                                       nextcloud-aio-domaincheck
...
```
4. Skipped this because the reverse proxy is not containerized
5. Not done yet because I did not think this would help if I do not change any step. Hmmm, now that I take a closer look into that `docker ps` output, the aio-domaincheck container seems to still use port 11443, so I will recreate both containers with identical ports.

**Comment by Kuchenmampfer** [comment]
Hmmmm,
the containers got restarted when I updated to the latest version. Now the `sudo docker ps` output looks much better:
```
bf77bc3ad48d   nextcloud/aio-domaincheck:latest   "/start.sh"              5 hours ago   Up 5 hours   0.0.0.0:11000->11000/tcp, :::11000->11000/tcp                                                                                                                                                                                       nextcloud-aio-domaincheck
85631eee8321   nextcloud/all-in-one:latest        "start.sh /usr/bin/s…"   5 hours ago   Up 5 hours   80/tcp, 8443/tcp, 0.0.0.0:8080->8080/tcp, :::8080->8080/tcp                                                                                                                                                                         nextcloud-aio-mastercontainer
```
But I still get that error, so my guess was wrong.

**Comment by szaimen** [comment]
Here are all checks that are being done: 
https://github.com/nextcloud/all-in-one/blob/1bb0fb70065174d1ef8fd4f56975bf8f3d61cdd1/php/src/Data/ConfigurationManager.php#L190-L235

Seems like it fails at the last one so all checks before that seem to work which is already pretty good...

**Reply by szaimen** [reply]
So basically 
1. it is a valid domain
2. a dns config is set
3. port 443 seems to be open for that domain
4. Now only the instance id does not seem to match which could be due to a wrong certificate, the reverse proxy not pointing to the correct port/server, some firewall in the way, and much more...

**Comment by Kuchenmampfer** [comment]
This makes me think that it is a certificate error:
![Screenshot_20220516_220851](https://user-images.githubusercontent.com/79256346/168674920-a103cd06-f3f6-4bf3-80a0-3b431f3a931d.png)
So I guess I have to learn about certificates in order to fix this. I am very new to them and even do not really know why I have a certificate for my other subdomains...
Do you have a suggestion where I should start to look or shall I just write my domain registrars (and vps providers) support?

**Reply by szaimen** [reply]
Yes looks like it. In theory caddy should automatically get a valid certificate from lets encrypt... did you already look at the caddy logs?

**Comment by Kuchenmampfer** [comment]
Nope, I can not find a hint where I can find them.

**Reply by szaimen** [reply]
Then you might need to google how to find the caddy logs...

**Reply by Kuchenmampfer** [reply]
Yes I did that. But nevermind, I had a typo in my journalctl command. Bruh

**Comment by Kuchenmampfer** [comment]
Ah yeah it failed because the ports (80 and 443) were already used by my mail server. So I will stop that and look through its reverse proxy setup what I have to change. Thank you very much for guiding me so far!
![Screenshot_20220516_225922](https://user-images.githubusercontent.com/79256346/168682108-f976cf7a-06bb-47c7-b13c-5ff16bd1be51.png)

**Comment by Kuchenmampfer** [comment]
Alright, I removed the mailserver (still need to figure out how to make it run behind caddy, but that does not belong here) and restarted everything. Now setting up nextcloud worked nicely up to this point:
![Screenshot_20220521_143313](https://user-images.githubusercontent.com/79256346/169651696-9816800c-7867-444d-b18b-afa95cda2839.png)
The Apache logs say they are waiting for the nextcloud container and its logs say the following many times:
```
Configuring Redis as session handler
touch: /mnt/ncdata/this-is-a-test-file: Permission denied
The www-data user doesn't seem to have access rights in /mnt/ncdata.
Did you maybe change the datadir and did forget to apply the correct permissions?
```
Creating `/mnt/ncdata` owned by `www-data` on the host system did not solve this. What can I do to get this fixed?

**Reply by szaimen** [reply]
Can you post the docker rung xommand that you used?

**Comment by Kuchenmampfer** [comment]
```
sudo docker run -it --name nextcloud-aio-mastercontainer --restart always -p 8080:8080 -e APACHE_PORT=11000 --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config --volume /var/run/docker.sock:/var/run/docker.sock:ro nextcloud/all-in-one:latest > nextcloud.log
```
This one, but I used `ctrl+C` afterwards to detach it from the shell because `ctrl+Z` did not do anything. Not sure if the error occured for the first time before or after that though.

**Reply by szaimen** [reply]
All right. This seems to be a hickup. Can you try to start from scratch by following https://github.com/nextcloud/all-in-one/tree/enh/665/reset-instance#how-to-properly-reset-the-instance or do you already have some data in place?

**Reply by szaimen** [reply]
Afterwards simply use the same command again that you used before to create the AIO instance.

**Comment by Kuchenmampfer** [comment]
Yaay, it works. Thank you very much for your help :)

**Reply by Kuchenmampfer** [reply]
Now I only need to figure out how to mark your reply as the answer haha
